lib/models/calculator.js

```diff
@@ -0,0 +1,182 @@
+
+    // Calculate results for each CQL statement
+    const resultsRaw = Calculator.executeEngine(
+      elm,
+      patientSource,
+      measureValueSets,
+      measure.main_cql_library,
+      mainLibraryVersion,
+      executionDateTime,
+      params
+    );
+
+    Object.keys(resultsRaw.patientResults).forEach((patientId) => {
+      let populationResults;
+      let episodeResults;
+      let populationRelevance;
+
+      // Parse CQL statement results into population values
+      measure.populations.forEach((population, populationIndex) => {
+        [populationResults, episodeResults] = Array.from(CalculatorHelpers.createPopulationValues(
+          measure,
+          population,
+          resultsRaw,
+          patientId,
+          observationDefs,
+          populationIndex
+        ));
+        if (populationResults) {
+          const result = CqmModels.IndividualResult();
+          result.set(populationResults);
+          if (episodeResults != null) {
+            // In episode of care based measures, episodeResults contains the population results
+            // for EACH episode.
+            result.episode_results = episodeResults;
+            if (Object.keys(episodeResults).length > 0) {
+              /* In episode of care based measures, episode_results contains the population results
+               * for EACH episode, so we need to build population_relevance based on a combonation
+               * of the episode_results. IE: If DENEX is irrelevant for one episode but relevant for
+               * another, the logic view should not highlight it as irrelevant
+               */
+              populationRelevance = ResultsHelpers.populationRelevanceForAllEpisodes(episodeResults);
+            } else {
+              // Use the patient based relevance if there are no episodes. This will properly set IPP or STRAT to true.
+              populationRelevance = ResultsHelpers.buildPopulationRelevanceMap(populationResults);
+            }
+          } else {
+            // Calculate relevance for patient based measure
+            populationRelevance = ResultsHelpers.buildPopulationRelevanceMap(populationResults);
+          }
+
+          // Build statement relevance mappings
+          const statementRelevance = ResultsHelpers.buildStatementRelevanceMap(
+            populationRelevance,
+            measure,
+            population,
+            populationIndex
+          );
+
+          result.extendedData.population_relevance = populationRelevance;
+          result.extendedData.statement_relevance = statementRelevance;
+
+          result.set(ResultsHelpers.buildStatementAndClauseResults(measure, resultsRaw.localIdPatientResultsMap[patientId], statementRelevance, !!options.doPretty));
+
+          // Populate result with info
+          result.patient = patientId;
+          result.measure = measure._id;
+          result.state = 'complete';
```

Comment:
I believe this is an artifact from the QDM calculator in Bonnie. You should be able to remove it.

Reply:
The individualResult schema has an enumerated state field